sdks/java/core/src/main/java/org/apache/beam/sdk/options/PortablePipelineOptions.java

```diff
@@ -50,6 +50,7 @@
   @Description(
       "Set the default environment type for running user code. "
           + "Possible options are DOCKER and PROCESS.")
+  @Nullable
```

Comment:
This option de facto defaults to DOCKER. Maybe we should consider making that more explicit by making that the `@Default` instead of null.

https://github.com/apache/beam/blob/8d2f37166a40f166bc9b7d47391e384246bfcb57/runners/core-construction-java/src/main/java/org/apache/beam/runners/core/construction/Environments.java#L100-L102

Reply:
Not always, as we now default to loopback execution in some cases. Could be addressed in a follow-up.

Reply:
> Not always, as we now default to loopback execution in some cases

But that's only for Python, isn't it?

Reply:
Shouldn't it be consistent and same for Java and Python SDKs?

Reply:
> Shouldn't it be consistent and same for Java and Python SDKs?

Ideally, yes. However, in Python we're setting LOOPBACK according to Flink/Spark master settings in (flink|spark)_runner.py, which don't have an equivalent in Java.